python/ray/serve/_private/http_state.py

```diff
@@ -325,6 +336,8 @@ def __init__(
         head_node_id: str,
         cluster_node_info_cache: ClusterNodeInfoCache,
         grpc_options: Optional[gRPCOptions] = None,
+        proxy_actor_class: Type[HTTPProxyActor] = HTTPProxyActor,
+        soft_node_affinity: bool = False,
```

Comment:
Note: `proxy_actor_class` and `soft_node_affinity` are now injected so we can use different setups in the test